### PR TITLE
Update approvers with SIGs and SMEs

### PR DESCRIPTION
**Pull request by fabiand**
**What this PR does / why we need it**:
Expand the concept of approvers by expanding SIGs and introducing SMEs.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
Please review and comment on https://docs.google.com/presentation/d/1_wqDLiY9X6iV94QWAtROiNR54nBqcThXb76Uhj_Qq3E/edit#

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: OWNERS
Comment (dhiller) [inline OWNERS:19]: How about adding an entry for `automation` folder?
```suggestion
  "automation/.*":
    approvers:
      - sme-buildsystem
  "hack/.*|tests/.*|.*_test.go":
```

File: OWNERS_ALIASES
Comment (dhiller) [inline OWNERS_ALIASES:41]: Maybe add David here, since he does the release?
```suggestion
      - davidvossel

```

File: OWNERS_ALIASES
Comment (iholder101) [inline OWNERS_ALIASES:11]: It's `iholder101` :)

File: OWNERS
Comment (fabiand) [reply OWNERS:19]: God idea!

File: OWNERS_ALIASES
Comment (fabiand) [reply OWNERS_ALIASES:43]: Yes, sounds fair ... although I wonder if it should then be build-system-n-releases

File: OWNERS_ALIASES
Comment (fabiand) [reply OWNERS_ALIASES:11]: Fixed, thanks!

File: pkg/OWNERS
Comment (dominikholler) [inline pkg/OWNERS:19]: Would a sme focused on `pkg/virtctl`  be helpful?

Comment (fabiand) [reply pkg/OWNERS:19]: Fixed

**Review by dhiller — APPROVED** [review]
Looks good to me!

File: OWNERS_ALIASES
Comment (lyarwood) [inline OWNERS_ALIASES:5]: supernit - A quick statement here about the differences between SMEs and SIGs here might be useful.

**Review by lyarwood — APPROVED** [review]
/lgtm

I'm just a reviewer at the moment but this looks okay to me, some quick supernits listed if you want to address them.

File: OWNERS_ALIASES
Comment (0xFelix) [reply OWNERS_ALIASES:5]: +1

**Review by 0xFelix — APPROVED** [review]
Nothing to add other than I have to agree with Lee's nit. LGTM.

/lgtm

**Comment by vladikr** [comment]
I'm fine with this approach.
/lgtm 
/cc @rmohr @davidvossel

**Comment by lyarwood** [comment]
This isn't directly related to this PR but given our move to longer release cycle windows with `v0.59.0` did we also want to think about potentially allowing another group of approvers from the older `release-.*` branches? This approach worked well for [OpenStack](https://docs.openstack.org/project-team-guide/stable-branches.html) in the past and allowed bugfixes etc flowing through the older branches while most approvers were focused on main/master/HEAD development.

**Comment by iholder101** [comment]
/lgtm

This is a huge step forward IMO.
As we need more approvers, I think that we should push this forward.

Are there any objections for approving this?
@vladikr @rmohr @davidvossel @xpivarc @acardace @dhiller

**Review by dhiller — APPROVED** [review]
/approve

I'd say this is good to go in. Waiting a day or two for others to chime in here.

/hold

If no objections occur, I will remove the hold on Thursday afternoon.

**Comment by xpivarc** [comment]
/hold
I would like to see a majority of maintainers taking a vote on this as this is a project structure change.  I have one suggestion here but I need time to verify if what I am going to propose works.

**Comment by iholder101** [comment]
> /hold I would like to see a majority of maintainers taking a vote on this as this is a project structure change. I have one suggestion here but I need time to verify if what I am going to propose works.

This PR is now opened for about 3.5 months.
To be completely honest, my opinion is that we had more than enough time for anyone to chime in.

I agree with @dhiller. We shouldn't wait for everyone to join the discussion, but set a reasonable deadline for any objections to be heard. 3.5 months of deadline is more than enough IMO.

The fact that you re-hold the PR at the day of the deadline effectively saying it would be holded again for an unknown amount of time is very concerning.

Just my opinion.

**Comment by fabiand** [comment]
/hold

… due to good thoughts from several people (which became aware of this PR after I rebased it a coupel of times)

**Comment by fabiand** [comment]
@iholder101 @dhiller +1

I think one concern that got some more attention is the fact of "global approval rights".
Let me reowkr this PR and follow up.

Thanks for all your inputs for now.

**Comment by fabiand** [comment]
#9453 is taking a softer approach by only laying the foundation, but not yet making the step to change ownership.

**Pull request closed without merging**
